projects/plugins/crm/includes/ZeroBSCRM.Database.php

```diff
@@ -879,6 +880,33 @@ function zeroBSCRM_createTables(){
   COLLATE = ".$collation.";"; 
   zeroBSCRM_db_runDelta($sql);
 
+	// Add table to store automation workflows.
+	// phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+	$sql = 'CREATE TABLE IF NOT EXISTS ' . $ZBSCRM_t['automation-workflows'] . '(
+	`id` INT NOT NULL AUTO_INCREMENT,
+	`zbs_site` INT NOT NULL,
+	`zbs_owner` INT NOT NULL,
+	`name` VARCHAR(255) NOT NULL,
+	`description` VARCHAR(600) NULL DEFAULT NULL,
+	`category` VARCHAR(255) NOT NULL,
+	`triggers` JSON NOT NULL,
```

Comment:
**General context:**

The JSON data type isn't super common in the WordPress world. There tends to be a trend to serialize the data instead.

[WordPress.org server requirements](https://en-gb.wordpress.org/about/requirements/) mentions both MySQL and MariaDB:

> [MySQL](https://www.mysql.com/) version 5.7 or greater, OR [MariaDB](https://mariadb.org/) version 10.3 or greater.

**MySQL / MariaDB differences**

With the general trend to serialize arrays/objects combined with the lack of true support in MariaDB made me go with `LONGTEXT` instead of `JSON`: https://mariadb.com/kb/en/json-data-type/

Reply:
Yes, JSON type is a LONGTEXT for both, but with JSON we could perform some specific queries and also the internal validation they perform.

Using LONGTEXT will work anyway, for now, we will not perform any special query with the JSON data there. So fine with using `LONGTEXT`.